app/assets/js/app.js

```diff
@@ -593,30 +601,47 @@ app.provider('k8sApi',
                this.$get = ["$http", "$q", function($http, $q) {
                  var api = {};
 
-                 api.getUrlBase = function() { return urlBase + '/namespaces/' + _namespace; };
+                 api.getUrlBase = function() { return urlBase; };
 
-                 api.getPods = function(query) { return _get($http, api.getUrlBase() + '/pods', query); };
+                 api.getNamespacedUrlBase = function() { return urlBase + '/namespaces/' + _namespace; };
+
+                 api.getPods = function(query) { return _get($http, api.getNamespacedUrlBase() + '/pods', query); };
 
                  api.getNodes = function(query) { return _get($http, urlBase + '/nodes', query); };
 
                  api.getMinions = api.getNodes;
 
-                 api.getServices = function(query) { return _get($http, api.getUrlBase() + '/services', query); };
+                 api.getServices = function(query) { return _get($http, api.getNamespacedUrlBase() + '/services', query); };
 
                  api.getReplicationControllers = function(query) {
-                   return _get($http, api.getUrlBase() + '/replicationcontrollers', query)
+                   return _get($http, api.getNamespacedUrlBase() + '/replicationcontrollers', query)
                  };
 
-                 api.getEvents = function(query) { return _get($http, api.getUrlBase() + '/events', query); };
+                 api.getEvents = function(query) { return _get($http, api.getNamespacedUrlBase() + '/events', query); };
 
                  return api;
                }];
              })
     .config(["k8sApiProvider", "ENV", function(k8sApiProvider, ENV) {
       if (ENV && ENV['/'] && ENV['/']['k8sApiServer']) {
-        k8sApiProvider.setUrlBase(ENV['/']['k8sApiServer']);
+        var base = ENV['/']['k8sApiServer']
+
+        if (base[0] == '/' ) {
+          // try to find the k8sApiServer location in the current path, falling
+          // back to <k8sApiServer> if not found:
+          // e.g.: /some/prefix/api/v1/proxy/.../kube-ui => /some/prefix/api/v1
+          //       /kube-ui                              => /api/v1
+          var re = new RegExp('^.*?' + regexEscape(base));
+          var result = re.exec(window.location.pathname)
```

Comment:
Applying a regex to a URL always makes me nervous. Would it be reasonable to instead embed a hidden element to be rewritten (e.g. <a href="/" style="display:none;">), or parse an existing element src (like the logo), and get the source from that?